### PR TITLE
Added quick start guide for Java.

### PR DESCRIPTION
**Pull request by binakot**
Related to #40

### CODE REVIEW

**Comment by jonatas** [comment]
Thank you very much @binakot! I'll try to review it during this week.

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:225]:
```suggestion
Next you can formulate the `SELECT` statement to convert the table we created in Step 1 into a hypertable. 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:226]:
```suggestion
Note that you must specify the table name to convert to a hypertable 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:234]:
```suggestion
### Step 3: Execute statements from step 1 and step 2
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:238]:
```suggestion
Now you can bring it all together by executing the statement from step 1, 
then executing the statement from step 2 
and committing your changes to the database:
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:287]:
```suggestion
Congratulations, you've successfully created a hypertable in your TimescaleDB database using Java.
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:291]:
```suggestion
Here's a typical pattern you could use to insert some data into a table. 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:292]:
```suggestion
In the example below, insert the relational data from list of sensors, 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:295]:
```suggestion
First, open a connection to the database, 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:296]:
```suggestion
then using prepared statements formulate the `INSERT` SQL statement 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:297]:
```suggestion
and then execute that statement:
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:314]:
```suggestion
You can insert a batch of rows into TimescaleDB in a couple of different ways. 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:435]:
```suggestion
Congratulations, you've successfully inserted data into TimescaleDB using Java.
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:441]:
```suggestion
First, define the SQL query you'd like to run on the database. 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:443]:
```suggestion
It returns the average CPU values for every 15 minute interval for sensors with specific type and location.
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:458]:
```suggestion
Now you can execute the query with the prepared statement and read out the result set 
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:602]:
```suggestion
Congratulations, you've successfully executed a query on TimescaleDB using Java and PostgreSQL JDBC.
```

File: timescaledb/quick-start/java.md
Comment (Loquacity) [inline timescaledb/quick-start/java.md:606]:
```suggestion
Now that you're able to connect, read, and write to a TimescaleDB instance from your Java application, 
```

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:5]: Sure, it's just a copy-paste from other guides (golang, ruby, python, node) :)

**Comment by binakot** [comment]
All fixes are done! 

I think it is worth reviewing the rest of the guides, since most of the text was copied from them :smile:

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:81]: For some reason, this didn't work by default.

```bash
javac -cp postgresql-42.2.20.jar com/timescale/java/Main.java
com/timescale/java/Main.java:9: error: cannot find symbol
        DriverManager.drivers().forEach(System.out::println);
                     ^
  symbol:   method drivers()
  location: class DriverManager
1 error
```

Comment (jonatas) [reply timescaledb/quick-start/java.md:81]: I'm trying a few attempts and I'm a bit lost on the java compiler flags.

```bash
java -cp *.jar com/timescale/java/Main                16:02:44
Error: Could not find or load main class com.timescale.java.Main
javac -cp *.jar com/timescale/java/Main.java          16:02:47
com/timescale/java/Main.java:9: error: cannot find symbol
        DriverManager.drivers().forEach(System.out::println);
                     ^
  symbol:   method drivers()
  location: class DriverManager
1 error
```

Comment (jonatas) [reply timescaledb/quick-start/java.md:81]: I have [this jar](https://jdbc.postgresql.org/download/postgresql-42.2.20.jar) in my folder.

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:209]: Maybe we could avoid the repetition of `Formulate ... statement for your hypertable` in all the subtitles?

I'd call like:

```md
### Step 1: Create `sensor_data` table with timestamp
...
### Step 2: Call `create_hypertable` for `sensor_data` table
...
### Step 3: Wrap previous steps into a java application
...
```

Or without emphasis in the terms:

```md
### Step 1: Create sensors data table
...
### Step 2: Create hypertable for sensors data
...
### Step 3: Execute previous steps from your Java code
```

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:225]:
```suggestion
Next, you can formulate the `SELECT` statement to convert the table we created in Step 1 into a hypertable.
```

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:231]:
```suggestion
SELECT create_hypertable('sensor_data', 'time');
```

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:275]:
```suggestion
                        value DOUBLE PRECISION
```

File: timescaledb/quick-start/java.md
Comment (jonatas) [inline timescaledb/quick-start/java.md:602]: Missing the 🎉 Emoji 😄

**Review by jonatas — CHANGES_REQUESTED** [review]
Thank you very much for the Java guidance here 🚀  🙇 

Today I revived my past working with Java :neckbeard: 

I've made a few suggestions and get stuck because of the library incompatibility in my mac.

Looking forward to checking if the jar is the wrong version. I made a few suggestions that you can ignore if you think it does not make sense. I think we have a single `value` column that is easy to follow as the 2 columns just have the same behavior.

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:34]: For simplicity there is no directory structure, just a single folder with `Main.java` file and postgres jdbc jar artefact. You don't have to create directory structure like the package path for single file source code. Starting in Java 11, we can use the following command to execute a single-file program: `java Main.java` ignoring 2 steps way: `javac Main.java` + `java Main`.

Comment (binakot) [reply timescaledb/quick-start/java.md:34]: What version of JDK are you using? What is output of `java -version`?

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:51]: You created folder structure related to package path. It's not required for single file source code.

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:81]: I think this problem is related to the same thing: you tried to create folder structure com/timescaledb/java that should be skipped and replace just with folder contains 2 files: Main.java and postgres jdbc jar.

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:209]: Sure, I totally agree. Almost all text is copy-paste from other quick-starts. I just tried to repeat everything from already exist guides but on java language :)

**Comment by binakot** [comment]
> Thank you very much for the Java guidance here 🚀 🙇
> 
> Today I revived my past working with Java :neckbeard:
> 
> I've made a few suggestions and get stuck because of the library incompatibility in my mac.
> 
> Looking forward to checking if the jar is the wrong version. I made a few suggestions that you can ignore if you think it does not make sense. I think we have a single `value` column that is easy to follow as the 2 columns just have the same behavior.

Thanks for the review. I'm all for it! I agree with the comments. I just tried to transfer examples from other guides. I didn’t add anything of my own, only the language was replaced to Java ;)

I will do my best!

**Comment by binakot** [comment]
About my setup and how I run it:

```bash
~/Projects/github/timescale-docs/temp/# uname -a
Darwin iMac-Ivan.local 19.6.0 Darwin Kernel Version 19.6.0: Mon Apr 12 20:57:45 PDT 2021; root:xnu-6153.141.28.1~1/RELEASE_X86_64 x86_64

~/Projects/github/timescale-docs/temp/# java -version           
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment (build 16+36-2231)
OpenJDK 64-Bit Server VM (build 16+36-2231, mixed mode, sharing)

~/Projects/github/timescale-docs/temp/# ls    
Main.java   postgresql-42.2.20.jar

~/Projects/github/timescale-docs/temp/# java -cp *.jar Main.java
org.postgresql.Driver@50eac852
```

Here is test java file:

```java
package com.timescale.java;

import java.sql.DriverManager;

public class Main {

    public static void main(String... args) {
        DriverManager.drivers().forEach(System.out::println);
    }
}
```

The TimescaleDB instance I run with docker-compose file:

```yaml
version: '3.7'

services:
  postgres:
    image: timescale/timescaledb:latest-pg12
    container_name: postgres
    restart: unless-stopped
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - 5432:5432
```

That's all and everything works on my macos, linux and even windows 10 😸 

Can you try to repeat that on your machine?)

File: timescaledb/quick-start/java.md
Comment (binakot) [reply timescaledb/quick-start/java.md:602]: Before that, there was an exclamation mark at the end of the sentence, but @Loquacity removed it. I thought that excessive impression is not welcome. Although, of course, I would add emoji 😆

**Comment by binakot** [comment]
@jonatas  All recommendations are implemented :octocat:

File: timescaledb/quick-start/java.md
Comment (jonatas) [reply timescaledb/quick-start/java.md]: Yes! they're full of issues and let's try to improve from here! Thanks again for your patience!

File: timescaledb/quick-start/java.md
Comment (jonatas) [reply timescaledb/quick-start/java.md:34]: Java 11 😬 

here is my java version:


```
java -version                                                                         20:21:43
java version "1.8.0_231"
Java(TM) SE Runtime Environment (build 1.8.0_231-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.231-b11, mixed mode)```

**Review by jonatas — APPROVED** [review]
After updating everything worked as expected!

Maybe we should just confirm the minimum java version to reproduce the steps.

After using a simple `brew install --cask adoptopenjdk16` the sky is clear and the pg drivers are working 🌞 

Tested all steps! everything worked as described! Thanks a lot for all the effort here @binakot!

File: timescaledb/quick-start/java.md
Comment (Loquacity) [reply timescaledb/quick-start/java.md]: I'm a tech writer, I thought the "Congratulations" was probably exuberant enough 😂  I don't mind what we choose, though, as long as it's consistent everywhere 👍

**Comment by binakot** [comment]
> After updating everything worked as expected!
> 
> Maybe we should just confirm the minimum java version to reproduce the steps.
> 
> After using a simple `brew install --cask adoptopenjdk16` the sky is clear and the pg drivers are working 
> 
> Tested all steps! everything worked as described! Thanks a lot for all the effort here @binakot!

Awesome!

I already added additional information about Java version:

```
All code is presented for Java 16 and above. 
If you are working with older JDK versions, use legacy coding techniques.
```

And also:

```
You don't have to create directory structure `./com/timescale/java` similar to package path in source file.
You should just create a single java file in empty folder and run `java Main.java` from it.
```

Is it enough?

**Comment by jonatas** [comment]
Yes! definitely @binakot! Thanks for the contribution! Let's sync with @ryanbooz and probably we're ready to merge!

**Comment by binakot** [comment]
Do I need to do anything else or are we waiting for the final review by @ryanbooz ? 🤔

**Comment by Loquacity** [comment]
> Do I need to do anything else or are we waiting for the final review by @ryanbooz ?

I'm happy to merge, up to you if you want more reviews :)

**Comment by binakot** [comment]
> Yes! definitely @binakot! Thanks for the contribution! Let's sync with @ryanbooz and probably we're ready to merge!

@Loquacity, I think we are ready to merge, but @jonatas said to sync with @ryanbooz. 
If it's not mandatory, let's merge & deploy :octocat: 😄

File: timescaledb/quick-start/java.md
Comment (ryanbooz) [inline timescaledb/quick-start/java.md:614]:
```suggestion
[timescaledb-hypertable]: /overview/core-concepts/hypertables-and-chunks/
```

**Review by ryanbooz — CHANGES_REQUESTED** [review]

File: timescaledb/quick-start/java.md
Comment (ryanbooz) [reply timescaledb/quick-start/java.md:614]: Just correct this one link but otherwise great! You've done a great job adapting what we had. As @Loquacity mentioned, it's obvious that all of these Quick Starts are in need of some attention to build further.  This is a good reminder to do that!

Thanks again for the contribution!!

**Review by ryanbooz — APPROVED** [review]

**Pull request merged**
